File: templates/template carga incron.R
################ NO BORRAR #######################
source("/opt/lib_anasac/anasac/init.R")
################ NO BORRAR #######################


tryCatch({
  # en general... proceso identificado por tabla de destino
  param = c(
    db = 'SANDBOX',
    schema = 'dbo',
    table = 'tabla_borrar'
  )
  param["proc"] = paste(param,collapse = '.')
  
  # system('./setup_rclone.sh',intern = F)
  library(AzureStor)
  # library(readxl)
  
  args = commandArgs(trailingOnly=TRUE)
  archivo = args[1]
  # archivo + "ejemplo.xlsx"
  print(paste("archivo:", archivo))
  
  file_path = paste0("files/",archivo)
  print(paste("full path:", file_path))
  
  #La hoja correcta de ABO es DIARIO, debe comenzar a leer desde la linea 1 
  
  data <- read_excel(file_path)
 
  ########manipular data
  
  
  
  #######guardar
  
  #Atenea
  sqlExecute(db_atenea,"delete from @proc@ where mes_carga = '@mes@'", param)
  
  table_id <- Id(catalog = param[['db']], schema = param[['schema']], table = param[['table']])
  sqlWriteTable(db_atenea,data, table_id)
  
  
  ######### backups en blob
  blob_key = key_vault$get_value("azure_analytics300_key")
  blob_endpoint = key_vault$get_value("azure_analytics300_endpoint")
  blob_endp <- blob_endpoint(blob_endpoint,key=blob_key)
  cont <- blob_container(blob_endp, "backup-carga-archivos")
  blob_name = paste0(param[['proc']],"/",Sys.time(), " ",archivo)
  storage_multiupload(cont, archivo, blob_name)
  
  
  file.remove(file_path)
  write_log(param[['proc']], mensaje = archivo, blob_name)
  
},error=function(e) {
  write_log(param[['proc']],exit_code = -1,mensaje = paste(e$message,collapse = " "))
  print(e$message)}
)

File: templates/template carga incron.py
#!/bin/python3
################ NO BORRAR #######################
import os
import sys

while not any([x in  [".git","files","project.Rproj"] for x in os.listdir()]):
    os.chdir("..")
# print("Working Directory: " + os.getcwd())

if not os.getcwd() in sys.path:
    sys.path.append(os.getcwd())
    
if not '/opt/lib_anasac' in sys.path:
    sys.path.append('/opt/lib_anasac')

import anasac
################ NO BORRAR #######################

def load_file(archivo):
  try:
    
    params = {
      'db' : 'SANDBOX',
      'schema' : 'dbo',
      'table' : 'borrar'
    }
    params["proc"] = '.'.join(params.values())
    
    import pandas as pd
    # pip3 install azure-storage-blob --user
    from azure.storage.blob import BlobServiceClient, BlobClient, ContainerClient
    
    file_path = './files/' + archivo
    data = pd.read_excel(file_path)
    #preparar la tabla
    
    
    
    #limpar antes de cargar
    con = anasac.db_atenea(params['db'])
    anasac.sql_execute_query(con,'truncate table {}'.format(params["proc"]))
    
    # ingresar los datos,hay 2 formas
    # pandas
    data.to_sql(params["proc"],con_atenea,schema=params['schema'],if_exists="append",index=False)
    # bulk insert
    anasac.atenea_bulk_insert_dataframe(data,params["proc"])

    # queda pendiente respandar en blob
    blob_string = anasac.key_vault.get_value('azure_analytics300_string')
    blob_service_client = BlobServiceClient.from_connection_string(blob_string)
    container_name = "backup-carga-archivos"
    blob_file = params["proc"]+'/'+ str(anasac.datetime.datetime.now()) + " " + archivo

    blob_client = blob_service_client.get_blob_client(container=container_name, blob=blob_file)
    print("\nUploading to Azure Storage as blob:\n\t" + file_path)
    with open(file_path, "rb") as data:
        blob_client.upload_blob(data)
    
    #limpiar y log
    os.remove(file_path)
    con.close()
    anasac.write_log(tabla,exit_code = 0,mensaje = blob_file)
    
  except Exception as e:
    print(e)
    anasac.write_log(tabla,exit_code = -1, mensaje = str(e))



if __name__ == '__main__':
  args = sys.argv[1:]
  archivo = args[0]
  # archivo = "sap.sta.faglflext.csv"
  # archivo = "SAP.sta.MARA.csv"
  load_file(archivo)
